store/sqlstore/post_store.go

```diff
@@ -474,7 +474,23 @@ func (s *SqlPostStore) getPostWithCollapsedThreads(id, userID string, extended b
 
 		return nil, errors.Wrapf(err, "failed to get Post with id=%s", id)
 	}
-	return s.prepareThreadedResponse([]*postWithExtra{&post}, extended, false)
+
+	var posts []*model.Post
+	_, err = s.GetReplica().Select(&posts, "SELECT * FROM Posts WHERE Posts.RootId = :RootId AND DeleteAt = 0", map[string]interface{}{"RootId": id})
+	if err != nil {
+		return nil, errors.Wrap(err, "failed to find Posts")
+	}
+
+	list, err := s.prepareThreadedResponse([]*postWithExtra{&post}, extended, false)
+	if err != nil {
+		return nil, err
+	}
+	for i := range posts {
+		idx := i
+		list.AddPost(posts[idx])
+		list.AddOrder(posts[idx].Id)
+	}
```

Comment:
```suggestion
	for _, p := range posts {
	        list.AddPost(p)
		list.AddOrder(p)
	}
```

0/10 non-blocking suggestion.